src/matrix-utils.ts

```diff
@@ -55,16 +55,23 @@ const SYNC_STORE_NAME = "element-call-sync";
 // (It's a good opportunity to make the database names consistent.)
 const CRYPTO_STORE_NAME = "element-call-crypto";
 
-function waitForSync(client: MatrixClient): Promise<void> {
+async function waitForSync(client: MatrixClient): Promise<void> {
+  // If there is a saved sync, wait for it to be restored before the first real sync,
+  // as the saved sync may be missing some state
+  let waitForSavedSync = !!(await client.store.getSavedSyncToken());
   return new Promise<void>((resolve, reject) => {
     const onSync = (
       state: SyncState,
       _old: SyncState | null,
       data?: ISyncStateData,
     ): void => {
       if (state === "PREPARED") {
-        client.removeListener(ClientEvent.Sync, onSync);
-        resolve();
+        if (waitForSavedSync) {
+          waitForSavedSync = false;
```

Comment:
This variable update seems to never be used. we write to it after reading it.

Reply:
This indeed gets used: if the client has a cached sync result, `waitForSavedSync` gets set to `true` and the client will sync twice: first for the cached sync, and next for a "real" sync.  Each sync will fire a `ClientEvent.Sync` event, meaning `onSync` will get called twice.  Since `onSync` forms a closure over `waitForSavedSync`, it can be used to track whether `onSync` had been called or not; only if it has been called once before should the listener be removed (so that the listener can fire for both syncs) & the promise be resolved.

I also confirmed that removing `waitForSync = false` has the UI hang, as that prevents the promise from being able to resolve & thus has the UI think that no sync has finished.

Reply:
In order to do that, the executor function would have to be async, since `getSavedSyncToken` is async.

However, eslint advises against using async executors, and has a rule to forbid them: https://eslint.org/docs/latest/rules/no-async-promise-executor

Reply:
Given the above, I will leave this function as it is, unless you have another recommendation for improving it.

Reply:
Sure that makes sense!